homeassistant/components/sensor/yweather.py

```diff
@@ -0,0 +1,196 @@
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=120)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Yahoo! weather sensor."""
+    from yahooweather import UNIT_C, UNIT_F
+
+    unit = hass.config.temperature_unit
+    woeid = config.get("woeid", None)
+    forecast = config.get("forecast", 0)
+
+    if woeid is None:
+        _LOGGER.critical("You need woeid from yahoo!")
+        return False
+
+    if forecast > 4:
+        _LOGGER.error("Yahoo! don't support %d days forcast!", forecast)
+        return False
+
+    # convert unit
+    if unit == TEMP_CELSIUS:
+        yunit = UNIT_C
+    elif unit == TEMP_FAHRENHEIT:
+        yunit = UNIT_F
+    else:
+        _LOGGER.error("Unit %s is not supported by yahoo!", unit)
+        return False
+
+    # for print HA style temp
+    SENSOR_TYPES["temperature"][1] = unit
+    SENSOR_TYPES["temp_min"][1] = unit
+    SENSOR_TYPES["temp_max"][1] = unit
+
+    # exists a yahoo api object for this woeid?
+    yahoo_api = YAHOO_WEATHER.get(woeid, None)
+
+    # api for woeid don't exist
+    if yahoo_api is None:
+        yahoo_api = YahooWeatherData(woeid, yunit)
+        if not yahoo_api.update():
+            _LOGGER.critical("Can't retrieve data from yahoo!")
+            return False
+        # store object to ram. It save api request count
+        YAHOO_WEATHER[woeid] = yahoo_api
+
+    dev = []
+    try:
+        for variable in config[CONF_MONITORED_CONDITIONS]:
+            if variable not in SENSOR_TYPES:
+                _LOGGER.error('Sensor type: "%s" does not exist', variable)
+            else:
+                dev.append(YahooWeatherSensor(yahoo_api, forecast, variable))
+    except KeyError:
+        pass
+
+    add_devices(dev)
+
+
+# pylint: disable=too-many-instance-attributes
+class YahooWeatherSensor(Entity):
+    """Implementation of an OpenWeatherMap sensor."""
+
+    def __init__(self, weather_data, forecast, sensor_type):
+        """Initialize the sensor."""
+        self._client = 'Weather'
+        self._name = SENSOR_TYPES[sensor_type][0]
+        self._type = sensor_type
+        self._state = STATE_UNKNOWN
+        self._unit = SENSOR_TYPES[sensor_type][1]
+        self._data = weather_data
+        self._forecast = forecast
+        self._code = None
+
+        # update data
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        if self._forecast == 0:
+            return '{} {}'.format(self._client, self._name)
+
+        return '{} {} {}'.format(self._client, self._name, str(self._forecast))
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._data.yahoo.Units.get(self._unit, self._unit)
```

Comment:
The Forecast sensor was extended to use given unit of measurements. The pressure is often in a very un-common unit.